server/lib/txgh-server/webhooks/gitlab/request_handler.rb

```diff
@@ -0,0 +1,67 @@
+require 'json'
+
+module TxghServer
+  module Webhooks
+    module Gitlab
+      class RequestHandler < TxghServer::Webhooks::Github::RequestHandler
+        def handle_request
+          handle_safely do
+            if gitlab_event == 'Push Hook'
+              if delete_event?
+                DeleteHandler.new(project, repo, logger, attributes).execute
+              else
+                PushHandler.new(project, repo, logger, attributes).execute
+              end
+            else
+              respond_with_error(400, 'Unexpected event type')
+            end
+          end
+        end
+
+        private
+
+        def attributes
+          unless gitlab_event == 'Push Hook'
+            return BlankAttributes.from_webhook_payload(payload)
+          end
+
+          if delete_event?
+            return DeleteAttributes.from_webhook_payload(payload)
+          end
+
+          PushAttributes.from_webhook_payload(payload)
+        end
+
+        def gitlab_event
+          request.env['HTTP_X_GITLAB_EVENT']
+        end
+
+        def delete_event?
+          payload.fetch('after', {}) == '0000000000000000000000000000000000000000'
```

Comment:
Wow, is the Gitlab API really this similar to the Github API?

Reply:
Mostly similar, but some of the differences are annoying. For example it doesn't have separate delete branch hook and it doesn't sign the webhooks